bindings/matrix-sdk-ffi/src/client.rs

```diff
@@ -1018,6 +1018,17 @@ impl Client {
         Ok(Arc::new(Room::new(room)))
     }
 
+    /// Leave a room by its room ID.
+    pub async fn leave_room_by_id(
```

Comment:
What's the purpose of this function, when you have `Room::leave()`? In other words, what are the conditions that make it necessary: a room that's out of sync shouldn't be present at all, and having "bad state locally" seems like a poor justification for API proliferation.

Reply:
I think you're right and we should focus on fixing the underlying issue in `RoomPreview` instead.